src/Plugin/EntityBrowser/Widget/Webdam.php

```diff
@@ -242,43 +378,55 @@ protected function prepareEntities(array $form, FormStateInterface $form_state)
   public function submit(array &$element, array &$form, FormStateInterface $form_state) {
     $assets = [];
     if (!empty($form_state->getTriggeringElement()['#eb_widget_main_submit'])) {
-      foreach ($form_state->getValue(['assets'], []) as $aid) {
-        if ($aid !== 0) {
-          $webdam_asset = $this->webdam->getAsset($aid);
-          $media_asset = Media::create([
-            'bundle' => 'webdam',
-            'uid' => '1',
-            'langcode' => 'en',
-            'status' => Media::PUBLISHED,
-            'name' => $webdam_asset->name,
-            'field_asset_id' => $aid,
-          ]);
-          $media_asset->save();
-          $assets[] = $media_asset;
-        }
-      }
+      $assets = $this->prepareEntities($form,$form_state);
     }
-    // $this->clearFormValues($element, $form_state);
     $this->selectEntities($assets, $form_state);
   }
 
-   /**
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+        'submit_text' => $this->t('Select assets'),
+      ] +
+      parent::defaultConfiguration();
+  }
+
+  /**
    * {@inheritdoc}
    *
    * TODO: Add more settings for configuring this widget
    *
    */
   public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildConfigurationForm($form, $form_state);
-    $form['multiple'] = [
-      '#type' => 'checkbox',
-      '#title' => $this->t('Accept multiple files'),
-      '#default_value' => $this->configuration['multiple'],
-      '#description' => $this->t('Multiple assets will only be accepted if the source field allows more than one value.'),
+    $bundle_info = $this->entityTypeBundleInfo->getBundleInfo('media');
+    $media_bundles = MediaBundle::loadMultiple(array_keys($bundle_info));
```

Comment:
`MediaBundle::loadMultiple()` uses a service that you can inject under the hood. Inject that via create() and don't call the static method like this.